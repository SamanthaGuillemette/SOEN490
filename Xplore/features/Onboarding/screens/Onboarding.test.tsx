import renderer from "react-test-renderer";
import Onboarding from "./Onboarding.screen";
import { render } from "@testing-library/react-native";

describe("<Onboarding />", () => {
  it("matches snapshot", () => {
    const tree = renderer.create(<Onboarding />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("<Onboarding />", () => {
  it("renders correctly", () => {
    const container = render(<Onboarding />);
    expect(container).toBeTruthy();
  });
});
