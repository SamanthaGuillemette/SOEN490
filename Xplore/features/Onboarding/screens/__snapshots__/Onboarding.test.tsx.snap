// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Onboarding /> matches snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "flex": 1,
        "paddingTop": "10%",
      },
      Object {
        "backgroundColor": "#FFFFFF",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "marginTop": "50%",
        },
        Object {
          "backgroundColor": "#FFFFFF",
        },
      ]
    }
  >
    <RCTScrollView
      decelerationRate="fast"
      horizontal={true}
      onScroll={[Function]}
      pagingEnabled={true}
      scrollEventThrottle={32}
      showsHorizontalScrollIndicator={false}
    >
      <View>
        <View
          style={
            Array [
              Object {
                "marginHorizontal": 30,
                "width": 690,
              },
              Object {
                "backgroundColor": "#FFFFFF",
              },
            ]
          }
        >
          <Image
            source={1}
            style={
              Object {
                "borderRadius": 15,
                "height": "40%",
                "marginBottom": 43,
                "width": "100%",
              }
            }
          />
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "poppins-bold",
                  "fontSize": 22,
                  "lineHeight": 33,
                },
                Object {
                  "color": "#585757",
                },
                Object {
                  "paddingHorizontal": 30,
                  "textAlign": "center",
                },
              ]
            }
          >
            Xplorify focuses on project-based learning
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "marginHorizontal": 30,
                "width": 690,
              },
              Object {
                "backgroundColor": "#FFFFFF",
              },
            ]
          }
        >
          <Image
            source={1}
            style={
              Object {
                "borderRadius": 15,
                "height": "40%",
                "marginBottom": 43,
                "width": "100%",
              }
            }
          />
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "poppins-bold",
                  "fontSize": 22,
                  "lineHeight": 33,
                },
                Object {
                  "color": "#585757",
                },
                Object {
                  "paddingHorizontal": 30,
                  "textAlign": "center",
                },
              ]
            }
          >
            Discover or create your own amazing project ideas
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "marginHorizontal": 30,
                "width": 690,
              },
              Object {
                "backgroundColor": "#FFFFFF",
              },
            ]
          }
        >
          <Image
            source={1}
            style={
              Object {
                "borderRadius": 15,
                "height": "40%",
                "marginBottom": 43,
                "width": "100%",
              }
            }
          />
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "poppins-bold",
                  "fontSize": 22,
                  "lineHeight": 33,
                },
                Object {
                  "color": "#585757",
                },
                Object {
                  "paddingHorizontal": 30,
                  "textAlign": "center",
                },
              ]
            }
          >
            Communicate with your team members in real-time
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "marginHorizontal": 30,
                "width": 690,
              },
              Object {
                "backgroundColor": "#FFFFFF",
              },
            ]
          }
        >
          <Image
            source={1}
            style={
              Object {
                "borderRadius": 15,
                "height": "40%",
                "marginBottom": 43,
                "width": "100%",
              }
            }
          />
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "poppins-bold",
                  "fontSize": 22,
                  "lineHeight": 33,
                },
                Object {
                  "color": "#585757",
                },
                Object {
                  "paddingHorizontal": 30,
                  "textAlign": "center",
                },
              ]
            }
          >
            Earn rewards upon project completion
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "marginHorizontal": 30,
                "width": 690,
              },
              Object {
                "backgroundColor": "#FFFFFF",
              },
            ]
          }
        >
          <Image
            source={1}
            style={
              Object {
                "borderRadius": 15,
                "height": "40%",
                "marginBottom": 43,
                "width": "100%",
              }
            }
          />
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "poppins-bold",
                  "fontSize": 22,
                  "lineHeight": 33,
                },
                Object {
                  "color": "#585757",
                },
                Object {
                  "paddingHorizontal": 30,
                  "textAlign": "center",
                },
              ]
            }
          >
            Stay engaged with the leaderboard
          </Text>
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginTop": 50,
                },
                Object {
                  "backgroundColor": "#FFFFFF",
                },
              ]
            }
          >
            <View
              style={
                Array [
                  undefined,
                  Object {
<<<<<<< HEAD
                    "backgroundColor": "#FFFFFF",
=======
                    "backgroundColor": "#F5F5F5",
>>>>>>> origin/develop
                  },
                ]
              }
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "poppins-regular",
                        "fontSize": 12,
                        "lineHeight": 18,
                      },
                      Object {
                        "color": "#6587FF",
                      },
                      Object {
                        "textDecorationLine": "underline",
                      },
                    ]
                  }
                >
                  Get Started
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 20,
            "paddingHorizontal": 40,
          },
          Object {
            "backgroundColor": "#FFFFFF",
          },
        ]
      }
    >
      <View
        style={
          Array [
            undefined,
            Object {
              "backgroundColor": "#FFFFFF",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "bottom": 0,
                "flexDirection": "row",
                "marginLeft": 5,
                "position": "relative",
              },
              Object {
                "backgroundColor": "#FFFFFF",
              },
            ]
          }
        >
          <View>
            <View
              style={
                Array [
                  Object {
                    "borderRadius": 10,
                    "height": 8,
                    "marginHorizontal": 5,
                    "width": 8,
                  },
                  Object {
<<<<<<< HEAD
                    "backgroundColor": undefined,
=======
                    "backgroundColor": "#E8E8E8",
>>>>>>> origin/develop
                  },
                ]
              }
            />
          </View>
          <View>
            <View
              style={
                Array [
                  Object {
                    "borderRadius": 10,
                    "height": 8,
                    "marginHorizontal": 5,
                    "width": 8,
                  },
                  Object {
<<<<<<< HEAD
                    "backgroundColor": undefined,
=======
                    "backgroundColor": "#E8E8E8",
>>>>>>> origin/develop
                  },
                ]
              }
            />
          </View>
          <View>
            <View
              style={
                Array [
                  Object {
                    "borderRadius": 10,
                    "height": 8,
                    "marginHorizontal": 5,
                    "width": 8,
                  },
                  Object {
<<<<<<< HEAD
                    "backgroundColor": undefined,
=======
                    "backgroundColor": "#E8E8E8",
>>>>>>> origin/develop
                  },
                ]
              }
            />
          </View>
          <View>
            <View
              style={
                Array [
                  Object {
                    "borderRadius": 10,
                    "height": 8,
                    "marginHorizontal": 5,
                    "width": 8,
                  },
                  Object {
<<<<<<< HEAD
                    "backgroundColor": undefined,
=======
                    "backgroundColor": "#E8E8E8",
>>>>>>> origin/develop
                  },
                ]
              }
            />
          </View>
          <View>
            <View
              style={
                Array [
                  Object {
                    "borderRadius": 10,
                    "height": 8,
                    "marginHorizontal": 5,
                    "width": 8,
                  },
                  Object {
<<<<<<< HEAD
                    "backgroundColor": undefined,
=======
                    "backgroundColor": "#E8E8E8",
>>>>>>> origin/develop
                  },
                ]
              }
            />
          </View>
        </View>
        <View>
          <View
            collapsable={false}
            style={
              Object {
                "backgroundColor": "#024089",
                "borderRadius": 10,
                "bottom": 0,
                "height": 8,
                "marginHorizontal": 5,
                "position": "absolute",
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
                "width": 18,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "paddingVertical": 10,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "poppins-regular",
                  "fontSize": 14,
                  "lineHeight": 21,
                },
                Object {
<<<<<<< HEAD
                  "color": undefined,
=======
                  "color": "#969696",
>>>>>>> origin/develop
                },
                undefined,
              ]
            }
          >
            SKIP
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
